monitoring/bin/deploy_monitoring_cluster.sh

```diff
@@ -96,400 +95,398 @@ helmRepoAdd prometheus-community https://prometheus-community.github.io/helm-cha
-      rooturl="https://$GRAFANA_FQDN/$GRAFANA_PATH" yq -i '.grafana."grafana.ini".server.root_url=env(rooturl)'     "$autogeneratedYAMLFile"
-
-      yq -i '.prometheus.ingress.hosts.[0]=env(PROMETHEUS_FQDN)'              "$autogeneratedYAMLFile"
-      yq -i '.prometheus.ingress.tls.[0].hosts.[0]=env(PROMETHEUS_FQDN)'      "$autogeneratedYAMLFile"
-      slashpath="/$PROMETHEUS_PATH" yq -i '.prometheus.ingress.path=env(slashpath)'                       "$autogeneratedYAMLFile"
-      slashpath="/$PROMETHEUS_PATH" yq -i '.prometheus.prometheusSpec.routePrefix=env(slashpath)'         "$autogeneratedYAMLFile"
-      exturl="https://$PROMETHEUS_FQDN/$PROMETHEUS_PATH" yq -i '.prometheus.prometheusSpec.externalUrl=env(exturl)'      "$autogeneratedYAMLFile"
-      slashpath="/$ALERTMANAGER_PATH" yq -i '.prometheus.prometheusSpec.alertingEndpoints.[0].pathPrefix=env(slashpath)' "$autogeneratedYAMLFile"
-
-
-      #NOTE: The tlsPromAlertingEnpointFile is no longer needed since sample
-      #      file already contains keys and we have been updated them above
-      tlsPromAlertingEndpointFile="$TMP_DIR/empty.yaml"
-
-      #Handle Prometheus datasource definition for Grafana
-      ##Copy and update YAML file
-      cp monitoring/tls/$grafanaDS  "$TMP_DIR/grafanaDS.yaml"
-      promurl="https://v4m-prometheus:9090/$PROMETHEUS_PATH" yq -i '.datasources.[0].url=env(promurl)'     "$TMP_DIR/grafanaDS.yaml"
-
-      ##Re-create configmap
-      kubectl delete cm -n $MON_NS --ignore-not-found grafana-datasource-prom-https
-      kubectl create cm -n $MON_NS grafana-datasource-prom-https --from-file  "$TMP_DIR/grafanaDS.yaml"
-      kubectl label  cm -n $MON_NS grafana-datasource-prom-https grafana_datasource=1 sas.com/monitoring-base=kube-viya-monitoring
-
-   fi
+if [ "$AUTOGENERATE_INGRESS" = "true" ]; then
+
+    if [ ! -f "$autogeneratedYAMLFile" ]; then
+        log_debug "Creating file [$autogeneratedYAMLFile]"
+        touch "$autogeneratedYAMLFile"
+    else
+        log_debug "File [$autogeneratedYAMLFile] already exists"
+    fi
+
+    if [ "$MON_TLS_PATH_INGRESS" != "true" ] && [ "$ROUTING" = "path" ]; then
+        log_error "Environment variable MON_TLS_PATH_INGRESS must be set to 'true' when ROUTING='path'."
+        exit 1
+    fi
+
+    ## tested with sample version: 0.2.2
+    ingressSampleFile="samples/ingress/${ROUTING}-based-ingress/monitoring/user-values-prom-operator.yaml"
+
+    #intialized the yaml file w/appropriate ingress sample
+    # shellcheck disable=SC2016
+    yq -i eval-all '. as $item ireduce ({}; . * $item )' "$autogeneratedYAMLFile" "$ingressSampleFile"
+
+    ALERTMANAGER_INGRESS_ENABLE="${ALERTMANAGER_INGRESS_ENABLE:-false}"
+
+    amIngressCert="${ALERTMANAGER_INGRESS_CERT}"
+    amIngressKey="${ALERTMANAGER_INGRESS_KEY}"
+    create_ingress_certs "$MON_NS" "alertmanager-ingress-tls-secret" "$amIngressCert" "$amIngressKey"
+
+    # Don't assign the variable to itself
```

Comment:
Comment?  Is this a reminder to make a change?
We used this approach to ensure the variable is exists and is set, i.e. if set, use the set value;  if not set, create it and use the default value. 
What's preferred alternative?

Reply:
Yea it was just a comment to remove that because it was a useless line. I'll remove the comments